src/plugins/intel_gpu/src/graph/include/matrix_nms_inst.h

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+///////////////////////////////////////////////////////////////////////////////////////////////////
+#pragma once
+#include <memory>
+#include <string>
+
+#include "intel_gpu/primitives/matrix_nms.hpp"
+#include "primitive_inst.h"
+
+namespace cldnn {
+
+using matrix_nms_node = typed_program_node<matrix_nms>;
+
+template <>
+class typed_primitive_inst<matrix_nms> : public typed_primitive_inst_base<matrix_nms> {
+    using parent = typed_primitive_inst_base<matrix_nms>;
+
+public:
+    typed_primitive_inst(network& network, const matrix_nms_node& node) : parent(network, node) {}
+
+    static layout calc_output_layout(const matrix_nms_node& node, const kernel_impl_params& impl_param);
+    static std::string to_string(const matrix_nms_node& node);
+
+    memory::ptr input_boxes_mem() const {
+        return dep_memory_ptr(0);
+    }
+    memory::ptr input_scores_mem() const {
+        return dep_memory_ptr(1);
+    }
+    memory::ptr second_output_mem() const {
+        return dep_memory_ptr(2);
+    }
+    memory::ptr third_output_mem() const {
```

Comment:
can we use semantic naming instead of third_output, second_output, like "seleted_indices_mem" for example

Reply:
done